src/datepicker/hijri/ngb-calendar-hijri.ts

```diff
@@ -0,0 +1,50 @@
+import {NgbDateHijri} from './ngb-date-hijri';
+import {NgbPeriod, NgbCalendar} from '../ngb-calendar';
+import {Injectable} from '@angular/core';
+
+@Injectable()
+export class NgbCalendarHijri extends NgbCalendar {
+  constructor(public dateModel: {new (...args: number[]): NgbDateHijri}) { super(); }
+  getDaysPerWeek() { return 7; }
+
+  getMonths() { return [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]; }
+
+  getWeeksPerMonth() { return 6; }
+
+  getNext(date: NgbDateHijri, period: NgbPeriod = 'd', number = 1) {
```

Comment:
It should be just `NgbDate` here, and it actually works for all methods not only this one